codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2ConnectionHandlerBuilder.java

```diff
@@ -152,11 +153,23 @@ protected long gracefulShutdownTimeoutMillis() {
      * Sets the graceful shutdown timeout of the {@link Http2Connection} in milliseconds.
      */
     protected B gracefulShutdownTimeoutMillis(long gracefulShutdownTimeoutMillis) {
+        if (gracefulShutdownTimeoutMillis < 0) {
+            throw new IllegalArgumentException("gracefulShutdownTimeoutMillis: " + gracefulShutdownTimeoutMillis +
+                                               " (expected: >= 0)");
+        }
         this.gracefulShutdownTimeoutMillis = gracefulShutdownTimeoutMillis;
         return self();
     }
 
     /**
+     * Sets the graceful shutdown timeout to be indefinite.
+     */
+    protected B gracefulShutdownIndefiniteWait() {
```

Comment:
I was originally trying to keep the existing API behavior the same, but allowing -1 is certainly more natural here. Changed.